File: web2-project/src/app/model/call.ts
import { Address } from "./address";
import { User } from "./user";

export enum Reason{
    NemaStruje,
    PostojiKvar,
    TreperenjeSvetla,
    PovratakStruje,
    DelimicnaStruja,
    ProblemiSNaponom
}

export class Call {
    public callID : number;
    public reason : Reason;
    public comment ?: string;
    public priority : number;
    public address : Address;
    public firstName ?: string;
    public lastName ?: string;

    constructor(reason : Reason, priority : number, address : Address, firstName?: string, lastName?: string, comment ?: string){
        this.reason = reason;
        this.comment = comment;
        this.priority = priority;
        this.address = address;
        this.firstName = firstName;
        this.lastName = lastName;
    }
}

File: Web2BackEnd/Web2BackEnd/Models/Call.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Web2BackEnd.Models
{
	public class Call : GenericModel
	{
		[Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
		public int CallID { get; set; }
		[Required]
		public Reason Reason { get; set; }
		public string Comment { get; set; }
		[Required]
		public int Priority { get; set; }
		[Required]
		public Address Address { get; set; }
		public string FirstName { get; set; }
		public string LastName { get; set; }
	}

	public enum Reason
	{
		NemaStruje,
		PostojiKvar,
		TreperenjeSvetla,
		PovratakStruje,
		DelimicnaStruja,
		ProblemiSNaponom
	}
}
